src/codemagic/tools/app_store_connect.py

```diff
@@ -148,6 +154,72 @@ def _delete_resource(self, resource_manager, resource_id: ResourceId, ignore_not
             else:
                 raise AppStoreConnectError(str(api_error))
 
+    @cli.action('list-testflight-builds',
+                BuildArgument.APPLICATION_ID_OPTIONAL,
+                BuildArgument.EXPIRED,
+                BuildArgument.NOT_EXPIRED,
+                BuildArgument.BUILD_ID_OPTIONAL,
+                BuildArgument.PRE_RELEASE_VERSION,
+                BuildArgument.PROCESSING_STATE,
+                BuildArgument.BUILD_NUMBER,
+                BuildArgument.ORDERING,
+                BuildArgument.REVERSE)
+    def list_testflight_builds(self,
+                              application_id: Optional[Union[str, ResourceId]] = None,
+                              expired: Optional[bool] = None,
+                              not_expired: Optional[bool] = None,
+                              build_id: Optional[Union[str, ResourceId]] = None,
+                              pre_release_version: Optional[str] = None,
+                              processing_state: Optional[BuildProcessingState] = None,
+                              build_number: Optional[str] = None,
+                              ordering: Optional[BuildOrdering] = None,
+                              reverse: Optional[bool] = False,
+                              should_print: bool = True) -> List[Build]:
+        """
+        List Testflight builds
+        """
+        try:
+            expired_value = get_binary_arguments_value(expired, not_expired)
+        except:
+            raise BuildArgument.NOT_EXPIRED.raise_argument_error('Using mutually exclusive flags "--expired" and "--not-expired"')
+
+        builds_filter = self.api_client.builds.Filter(
+            app=application_id,
+            expired=expired_value,
+            id=build_id,
+            processing_state=processing_state,
+            version=build_number,
```

Comment:
Is there any particular reason not to map CLI params directly to Apple's API filters. Here `version` vs `build_number` introduces somewhat of discrepancy.

Reply:
apple uses `build version` for build number and `build appstore version/build prerelease version` for build version. Did that to not confuse users with `build_version` and `pre_release_version`

Reply:
changed it to build_version_number [here](https://github.com/codemagic-ci-cd/cli-tools/pull/67/commits/71f2ca10806aa46be85d0feca84182780939f3d7#diff-caa1aae40f5faf0bf43c7eab968651cd6ce22681e07ed14d7ca18f2358fb781aR163)